src/Arcus.Messaging.Pumps.Abstractions/Resiliency/IMessagePumpCircuitBreaker.cs

```diff
@@ -33,8 +33,47 @@ public interface ICircuitBreakerEventHandler
         /// <summary>
         /// Notifies the application on a change in the message pump's circuit breaker state.
         /// </summary>
-        /// <param name="newState">The new circuit breaker state in which the message pump is currently running on.</param>
-        void OnTransition(MessagePumpCircuitState newState);
+        /// <param name="change">The change in the circuit breaker state for a message pump.</param>
+        void OnTransition(MessagePumpCircuitStateChange change);
+    }
+
+    /// <summary>
+    /// Represents a change event of the <see cref="MessagePumpCircuitState"/> in a <see cref="MessagePump"/>.
+    /// </summary>
+    public class MessagePumpCircuitStateChange
```

Comment:
Was thinking of making this and the `MessagePumpCircuitState` `struct`s, but didn't go through with for simplicity sake.